src/main/java/com/springweb/dv_spring_web_mongo/security/jwt/JwtFilter.java

```diff
@@ -0,0 +1,34 @@
+package com.springweb.dv_spring_web_mongo.security.jwt;
+
+import lombok.AllArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.filter.GenericFilterBean;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@AllArgsConstructor
+public class JwtFilter extends GenericFilterBean {
+
+    private JwtProvider jwtProvider;
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse res, FilterChain filterChain) throws IOException, ServletException {
+        String token = jwtProvider.resolveToken((HttpServletRequest) req);
+        if (token != null && jwtProvider.validateToken(token)) {
```

Comment:
You will parce the token two times here: first time in the method `jwtProvider.validateToken(token)`, second time in the method `jwtProvider.getUsername(token)`. Can you implement it more optimal? 
I sugges to merge two methods `JwtProvider.getUsername()` and `JwtProvider.validateToken()` -> `String JwtProvider.validateTokenAndGetUsername(String token)`. Calling of the method would include two actions: validate the token and return a username, return null if token is not valid.